Problem with `hiop_options` target

cmake fails with the following error message when cmake-ing with -DHIOP_DEVELOPER_MODE=ON

```
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.5") found components: doxygen dot 
-- Configuring Doxygen documentation
-- Configuring done
CMake Error: install(EXPORT "hiop-targets" ...) includes target "hiop_tpl" which requires target "hiop_options" that is not in any export set.
CMake Error: install(EXPORT "hiop-targets" ...) includes target "hiop_tpl" which requires target "hiop_warnings" that is not in any export set.
-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -140,6 +140,7 @@ target_compile_options(hiop_warnings
   -Wnull-dereference
   -Wdouble-promotion  # Warn on implicit conversion from
                       # float to double
+  -Wconversion
   )
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -158,6 +158,7 @@ endif()
 
 if(HIOP_DEVELOPER_MODE)
   target_link_libraries(hiop_tpl INTERFACE hiop_options hiop_warnings)
+  install(TARGETS hiop_options hiop_warnings EXPORT hiop-targets)
 endif()
 
 if(NOT (HIOP_USE_GPU AND HIOP_USE_HIP))

